src/OpalCompiler-Core/RBVariableNode.extension.st

```diff
@@ -13,7 +13,8 @@ RBVariableNode >> binding: aSemVar [
 
 { #category : #'*opalcompiler-core' }
 RBVariableNode >> isArg [
-
+	"This is a compiler specific function, requiring a binding.
+	If you're working on the AST, you should use probably isArgument"
```

Comment:
I've just checked at a non-semantically analysed AST. I did

```smalltalk
method parseTree.
```
And I got

![image](https://user-images.githubusercontent.com/708322/77782382-7c69de00-7057-11ea-80d8-baf044685038.png)

If you look, it does not use argument nodes. So actually `#isArg` and `#isArgument` are both compiler dependent. None of them is valid without a semantic analysis. The first one requires a binding, the second one requires the replacement of the node...

Reply:
On principle yes. But the compiler is not the only manipulating AST, and therefore let's call them "semantically analyses nodes".